erica/elster_xml/common/xml_conversion.py

```diff
@@ -0,0 +1,34 @@
+import copy
+from dataclasses import asdict
+
+import xmltodict as xmltodict
+
+from erica.elster_xml.common.basic_xml_data_representation import EXml
+
+
+class CustomDictParser(dict):
+    """
+    Parse the given object to a dict structure that xmltodict will interpret correctly.
+    Attributes starting with "xml_attr_" will be interpreted as XML attributes.
+    Attributes starting with "xml_only_text" will be treated as single text children of the XML element.
+    If an attribute has the value "None" it will be not included in the resulting dict.
+    """
+    def __init__(self, data):
+        new_data = copy.deepcopy(data)
+        for index, item in enumerate(data):
+            if item[0].startswith('xml_attr_'):
+                new_data[index] = (item[0].replace('xml_attr_', '@'), item[1])
+            if item[0].startswith('xml_only_text'):
+                new_data[index] = ('#text', item[1])
+            if item[1] is None:
+                new_data.remove(item)
+            if item[1] == {}:
+                new_data.remove(item)
+
+        super().__init__(new_data)
+
+
+def convert_object_to_xml(grundsteuer_object: EXml):
```

Comment:
One more suggestion: since we're already in OO territory, why not go all the way and have the object also implement the required behavior instead of just holding the state? You could move this logic to a method in `EXml` and call it directly on its instances.

Reply:
It's entirely up to you if you'd like to do this or not, but just to clarify: 

The only purpose of `EXml` is to be serialized into an XML. Having the logic that performs the serialization corresponds to the OO principle of encapsulation (i.e. putting data and related behavior together). Also it's not a bit rewrite, it literally amounts to moving the logic from xml_conversion.py to another file, adding a one-liner to `EXml` and possibly changing the imports in a couple of test files. 

But like I said, entirely up to you as far as I'm concerned, pick whichever design you find the most intuitive.